hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/ContentGenerator.java

```diff
@@ -48,38 +49,101 @@ public class ContentGenerator {
+
+  /**
+   * Issue Hflush after every write ( Cannot be used with Hsync ).
+   */
+  private final boolean hFlush;
 
-  ContentGenerator(long keySize, int bufferSize, int copyBufferSize) {
-    this.keySize = keySize;
-    this.bufferSize = bufferSize;
-    this.copyBufferSize = copyBufferSize;
+  ContentGenerator(Builder objectBuild) {
+    this.keySize = objectBuild.keySize;
+    this.bufferSize = objectBuild.bufferSize;
+    this.copyBufferSize = objectBuild.copyBufferSize;
+    this.hSync = objectBuild.hSync;
+    this.hFlush = objectBuild.hFlush;
     buffer = RandomStringUtils.randomAscii(bufferSize)
         .getBytes(StandardCharsets.UTF_8);
   }
 
-  /**
-   * Write the required bytes to the output stream.
-   */
+
   public void write(OutputStream outputStream) throws IOException {
-    for (long nrRemaining = keySize;
-         nrRemaining > 0; nrRemaining -= bufferSize) {
+    for (long nrRemaining = keySize; nrRemaining > 0;
+         nrRemaining -= bufferSize) {
       int curSize = (int) Math.min(bufferSize, nrRemaining);
       if (copyBufferSize == 1) {
         for (int i = 0; i < curSize; i++) {
           outputStream.write(buffer[i]);
+          flushOrSync(outputStream);
         }
       } else {
+        /**
+         * Write the required bytes to the output stream.
+         */
         for (int i = 0; i < curSize; i += copyBufferSize) {
-          outputStream.write(buffer, i,
-              Math.min(copyBufferSize, curSize - i));
+          outputStream.write(buffer, i, Math.min(copyBufferSize, curSize - i));
+          flushOrSync(outputStream);
         }
       }
     }
   }
 
+  private void flushOrSync(OutputStream outputStream) throws IOException {
+    if (outputStream instanceof FSDataOutputStream) {
+      if (hSync) {
+        ((FSDataOutputStream) outputStream).hsync();
+      } else if (hFlush) {
+        ((FSDataOutputStream) outputStream).hflush();
+      }
+    }
+  }
+
+  /**
+   * Builder Class for Content generator.
+   */
+  public static class Builder {
+
+    private long keySize = 1024;
+    private int bufferSize = 1024;
+    private int copyBufferSize = 1024;
+    private boolean hSync = false;
+    private boolean hFlush = false;
+
+    public Builder seyKeySize(long keysize) {
+      this.keySize = keysize;
+      return this;
+    }
+
+    public Builder setBufferSize(int buffersize) {
+      this.bufferSize = buffersize;
+      return this;
+    }
+
+    public Builder setCopyBufferSize(int copybuffersize) {
+      this.copyBufferSize = copybuffersize;
+      return this;
+    }
+
+    public Builder setHsyncData(boolean hsync) {
+      this.hSync = hsync;
+      return this;
+    }
+
+    public Builder setHflushData(boolean hflush) {
+      this.hFlush = hflush;
+      return this;
+    }
+
+    //Return the final constructed builder object
+    public ContentGenerator build() {
+      ContentGenerator contentgenerator = new ContentGenerator(this);
+      return contentgenerator;
```

Comment:
Builder's `build` method should verify that the object being constructed is valid.  Here `hflush` and `hsync` are not applicable at the same time.  Please add a check for that.

Nit: no need for local variable, return directly.

```suggestion
      return new ContentGenerator(this);
```